fetch.js

```diff
@@ -217,6 +217,11 @@
 
   function Request(url, options) {
     options = options || {}
+
+    if (url && url.match(/^([\w-]+:)?\/\/[^\/]+@/)) {
```

Comment:
> However in this case I think it's relatively trivial to detect whether username or password have been specified, and raise an error appropriately.

:smile: I disagree. This regex wouldn't match

- URLs with leading whitespace
- URLs with `+`, or `.` in the scheme.
- URLs with more than two `/` after the `:`
- Other valid, but pathological URLs.

Using `URL` would be much more robust.

> The worst thing that can happen is that this check fails and the XHR request made, but with user/password ignored because XHR never reads them from the URL string.

I'm not super familiar with the behavior of Fetch. If this isn't security sensitive, then a regex might be good enough.